source/agora/node/Validator.d

```diff
@@ -79,7 +79,11 @@ public class Validator : FullNode, API
 
     public override void start ()
     {
+        // Eliminate delay on first call
         this.startPeriodicDiscovery();
+        this.taskman.setTimer(5.seconds, &this.startPeriodicDiscovery,
+        Periodic.Yes);
+
```

Comment:
```suggestion
        // build the list of required quorum peers to connect to
        Set!PublicKey required_peer_keys;
        buildRequiredKeys(this.config.node.key_pair.address,
            this.config.quorum, required_peer_keys);

        import core.time;
        this.network.discover(required_peer_keys);  // avoid delay
        this.taskman.setTimer(5.seconds, { this.network.discover(required_peer_keys); },
            Periodic.Yes);
```

Reply:
And also remove the `startPeriodicDiscovery` function here too because it won't be needed  anymore.

Reply:
I thought it would be nice if the code was simple in the start function :)

Reply:
I'll test this code to see if it works.

`this.taskman.setTimer(5.seconds, { this.network.discover(required_peer_keys); },
  Periodic.Yes);
`

Reply:
> I thought it would be nice if the code was simple in the start function :)

Then move all of this code inside the `startPeriodicDiscovery` function. It doesn't make sense to have it both in `start` and a single function call in `startPeriodicDiscovery`

Reply:
The delegate function has a parameter :( 

```
source/agora/node/FullNode.d(160,30): Error: function `agora.common.Task.TaskManager.setTimer(Duration timeout, void delegate() dg, Periodic periodic = cast(Periodic)false)` is not callable using argument types `(Duration, void delegate(Set!(PublicKey) required_peer_keys = Set(null)), Periodic)`
source/agora/node/FullNode.d(160,30):        cannot pass argument `&this.network.discover` of type `void delegate(Set!(PublicKey) required_peer_keys = Set(null))` to parameter `void delegate() dg`
```

Reply:
Then wrap it in another delegate:

```D
this.taskman.setTimer(5.seconds, { this.network.discover(); },
            Periodic.Yes);
```